File: app/routers/attendance.py
from fastapi import APIRouter, Depends
from sqlalchemy.orm import Session

from app.dependencies import get_db
from app.models.attendance import Attendance
from app.schemas.attendance import AttendanceResponse, AttendanceUpsert

router = APIRouter(prefix="/attendance", tags=["Attendance"])


@router.post("/", response_model=AttendanceResponse)
def upsert_attendance(payload: AttendanceUpsert, db: Session = Depends(get_db)):
    row = (
        db.query(Attendance)
        .filter(Attendance.lesson_id == payload.lesson_id, Attendance.student_id == payload.student_id)
        .first()
    )
    if row:
        row.status = payload.status
        row.comment = payload.comment
    else:
        row = Attendance(**payload.model_dump())
        db.add(row)
    db.commit()
    db.refresh(row)
    return row


@router.get("/lesson/{lesson_id}", response_model=list[AttendanceResponse])
def list_attendance_for_lesson(lesson_id: int, db: Session = Depends(get_db)):
    return db.query(Attendance).filter(Attendance.lesson_id == lesson_id).all()

File: frontend/src/api/attendance.ts
/**
 * Файл src/api/attendance.ts:
 * Простое описание: этот файл управляет частью интерфейса или запросами к API.
 * Комментарий добавлен для прозрачности структуры фронтенда.
 */

import { apiRequest } from './client'

export type Attendance = {
  id: number
  lesson_id: number
  student_id: number
  status: string
  comment: string | null
}

export function getAttendanceByLesson(lessonId: number) {
  return apiRequest<Attendance[]>(`/attendance/lesson/${lessonId}`)
}

export function upsertAttendance(payload: Omit<Attendance, 'id'>) {
  return apiRequest<Attendance>('/attendance/', { method: 'POST', body: JSON.stringify(payload) })
}
